src/LinAlg/hiopVectorRajaPar.cpp

```diff
@@ -341,6 +341,30 @@ void hiopVectorRajaPar::copyFromStarting(int start_index, const hiopVector& src)
   rm.copy(this->data_dev_ + start_index, vdata, v.n_local_*sizeof(double));
 }
 
+/**
+ * @brief Copy `nv` elements from `start_index_in_v` at array `v` to this vector
+ *
+ * @param[in] start_index_in_v - position in v
+ * @param[in] v  - a raw array from which to copy into `this`
+ * @param[in] nv - how many elements of `v` to copy
+ *
+ * @pre Size of `v` must be >= nv.
+ * @pre start_index_in_v+nv <= size of 'v'
+ * @pre `this` is not distributed
+ *
+ * @warning Method casts away const from the `local_array`.
+ */
+void hiopVectorRajaPar::copy_from_starting_at(const double* v, int start_index_in_v, int nv)
+{
+  // If nothing to copy, return.
+  if(nv == 0)
+    return;
+
+  auto& rm = umpire::ResourceManager::getInstance();
+  double* vv = const_cast<double*>(v); // <- cast away const
+  rm.copy(data_dev_, vv + start_index_in_v, nv*sizeof(double));
```

Comment:
it's unusual that for a `copy` method to require the source to be non-const ...

Reply:
It is casting to/from void pointer somewhere I believe, that is why it fails for const pointers.

Reply:
I copy this line from [here](https://github.com/LLNL/hiop/blob/a7aa4e0dc030b3030c82c872308d2e753129cffe/src/LinAlg/hiopVectorRajaPar.cpp#L313). Other lines in the file also shows that non-const source is passed to umpire.
@pelesh @ashermancinelli  This is required by umpire, isn't it?

Reply:
`const void*` would not require the `const_cast`, but apparently is just a `void* ` ...

Reply:
https://github.com/LLNL/Umpire/blob/8d7fba7eaaefe194a90d1b26b8b7fb9138d485c3/src/umpire/ResourceManager.cpp#L383

Reply:
I remember we needed to cast away `const` to use Umpire copy.  Then again, that was three major versions ago. I'll check again.

CC @davidbeckingsale